api/src/main/thrift/org/apache/aurora/gen/api.thrift

```diff
@@ -169,6 +169,48 @@ enum Mode {
   RO = 2
 }
 
+/** Describes a parameter passed to docker cli */
+struct DockerParameter {
+  /** a parameter to pass to docker. (e.g. volume) */
+  1: string name
+  /** the value to pass to a parameter (e.g. /src/webapp:/opt/webapp) */
+  2: string value
+}
+
+struct DockerVolume {
+  /** The driver name on the host facilitate mounting the resource. */
+  1: string driver
+  /** The resource name that will serve as the source for the mount. */
+  2: string name
+  /** The specific options to pass on to the driver. */
+  3: optional list<DockerParameter> options
+}
+
+struct FileSecret {
+  /** The name field is used to lookup the secret source */
+  1: string name
+  /** The key field can be used to reference a single value within a secret containing arbitrary key-value pairs. */
+  2: optional string key
+}
+
+/** The type of volume mount */
+enum VolumeType {
+  UNKNOWN = 0
+  /** Represent container volume type 'HOST_PATH' requires isolator 'filesystem/linux' */
+  HOST_PATH = 1
+  /** Represent container volume type 'DOCKER_VOLUME' requires isolator 'docker/volume' */
+  DOCKER_VOLUME = 2
+  /** Represent container volume type 'FILE_SECRET' requires isolator 'volume/secret' */
+  FILE_SECRET = 3
+}
+
+union VolumeSource {
+  1: string hostPath
```

Comment:
I think by using this as a union, we can get away without using a VolumeType therefore shrinking our Thrift footprint and we reduce the error proneness of having to set the same value twice . (e.g. VolumeType== "X", VolumeSource == "X.value") 

In the code we can check isSet and use that as the "type" to choose the code path. Since Thrift won't allow more than one field in the union to be set we can safely depend on this. We use this same pattern in the SLA strategy https://github.com/apache/aurora/commit/f2acf53ff0d0cbb7b0c62f021cc55c9e45b32f9c#diff-dc70a52021b0286021f89682c3b0841bR241